flink-clients/src/main/java/org/apache/flink/client/LocalExecutor.java

```diff
@@ -77,11 +70,11 @@
 	// ------------------------------------------------------------------------
 
 	public LocalExecutor() {
-		this(null);
+		this(new Configuration());
 	}
 
 	public LocalExecutor(Configuration conf) {
-		this.baseConfiguration = conf != null ? conf : new Configuration();
+		this.baseConfiguration = checkNotNull(conf);
```

Comment:
I am used to using the `checkNotNull` but it is true that the `@Nonnull` is part of the whole java ecosystem so it may be worth using it instead. I would like to see other arguments as well and I am always up for uniformizing coding styles :)

Reply:
I think you can only omit the `checkNotNull()` calls when your build system verifies that the `@Nonnull` contract is actually followed. Outside code, that is calling your function might still pass in a `null`.

Reply:
@aljoscha thanks for your input. I agree that we could both make use of compile time constraint for internal development and place runtime checker when input may be from outside.